lib/pages/qr_scanner.dart

```diff
@@ -0,0 +1,112 @@
+                child: ChangeNotifierProvider(
+                    create: (_) => ScannerProvider(),
+                    builder: (context, snapshot) {
+                      return Consumer<ScannerProvider>(
+                          builder: (context, provider, snapshot) {
+                        return QRView(
+                            key: qrKey,
+                            overlay: QrScannerOverlayShape(
+                                borderColor: Theme.of(context).accentColor,
+                                borderRadius: 10,
+                                borderLength: 30,
+                                borderWidth: 10,
+                                cutOutSize: 250),
+                            onQRViewCreated: (controller) {
+                              controller.scannedDataStream
+                                  .listen((scanData) async {
+                                controller.pauseCamera();
+                                await provider.resolveDataScan(scanData);
+                                provider.submitQR();
+                                passToMainScreen = provider.result;
+                                ScaffoldMessenger.of(context).showSnackBar(
+                                  SnackBar(
+                                    backgroundColor:
+                                        Theme.of(context).primaryColor,
+                                    content: Row(
+                                      children: [
+                                        Spacer(),
+                                        Icon(
+                                          Icons.qr_code,
+                                          color: Colors.white,
+                                        ),
+                                        SizedBox(width: 10),
+                                        Text(
+                                          //Displays the format of the QR Code and the resuling text.
+                                          '${provider.format!.toUpperCase()}: ${provider.result!.toUpperCase()}',
+                                          style: TextStyle(
+                                              color: Colors.white,
+                                              fontSize: 16),
+                                        ),
+                                        Spacer()
+                                      ],
+                                    ),
+                                  ),
+                                );
+                              });
+                            },
+                            onPermissionSet: (ctrl, setValue) =>
+                                _onPermissionSet(context, ctrl, setValue));
+                      });
+                    })),
+          ],
+        ));
+  }
+
+//This is used to get the built in permission handler for the camera
+  void _onPermissionSet(
+      BuildContext context, QRViewController ctrl, bool setValue) {
```

Comment:
suggestion: I would change `setValue` to `hasPermission`. It is unclear if the value being set is having the permission or not at first glance.